### PR TITLE
Modular eventgrid

### PR DESCRIPTION
**Pull request by joheredi**

### CODE REVIEW

File: packages/rlc-common/src/helpers/nameConstructors.ts [hunk 1]
```diff
@@ -108,3 +108,16 @@ export function getImportModuleName(name: ModuleName, codeModel: RLCModel) {
...
 }
+
+export function getClientName(model: RLCModel) {
+  const clientName = model.libraryName;
+  let clientInterfaceName = clientName.endsWith("Client")
+    ? `${clientName}`
+    : `${clientName}Client`;
+
+  if (model.options?.isModularLibrary) {
+    clientInterfaceName = `${clientName.replace("Client", "")}Context`;
```
Comment (deyaaeldeen) [inline packages/rlc-common/src/helpers/nameConstructors.ts:119]: nitpick: is this equivalent?
```suggestion
    clientInterfaceName = clientName.replace("Client", "Context");
```

File: packages/typespec-test/test/openai_modular/generated/typespec-ts/package-files/package.json [hunk 2]
```diff
@@ -0,0 +1,133 @@
+{
```
Comment (qiaozha) [inline packages/typespec-test/test/openai_modular/generated/typespec-ts/package-files/package.json:1]: just out of curious why we have package.json both in the typespec-ts folder and in the  typespec-ts/package-file folder

Comment (joheredi) [reply packages/typespec-test/test/openai_modular/generated/typespec-ts/package-files/package.json:1]: This is temporary, we need to generate a propper package.json file that follows the modular configuration and that can take the azure-sdk-for-js flag to generate slightly different one. Right now the one in the folder generates what we need for azure-sdk-for-js: true. For now the root package.json file is the same as RLC, once we have this correctly we'll move the real one to the root and stop generating the RLC

File: packages/typespec-ts/src/modular/static/scripts/renameOutput.mjs [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+
+const __dirname = url.fileURLToPath(new URL('.', import.meta.url));
+const destinationPath = path.join(__dirname, "../dist/");
+
+const inputFile = path.join(destinationPath, "index.js");
+const inputMapFile = path.join(destinationPath, "index.js.map");
+
+if (fs.existsSync(inputFile) && fs.existsSync(inputMapFile)) {
+  fs.renameSync(inputFile, path.join(destinationPath, "index.cjs"));
+  fs.renameSync(inputMapFile, path.join(destinationPath, "index.cjs.map"));
```
Comment (qiaozha) [inline packages/typespec-ts/src/modular/static/scripts/renameOutput.mjs:16]: why we need to rename the output ?

**Review by qiaozha — COMMENTED** [review]
what does this file packages/typespec-ts/src/modular/buildStaticFiles.ts do ?

File: packages/typespec-test/test/openai_modular/generated/typespec-ts/package-files/package.json (hunk 2 above)
Comment (qiaozha) [reply packages/typespec-test/test/openai_modular/generated/typespec-ts/package-files/package.json:1]: Got it, thanks.

File: packages/typespec-ts/src/modular/static/scripts/renameOutput.mjs (hunk 3 above)
Comment (joheredi) [reply packages/typespec-ts/src/modular/static/scripts/renameOutput.mjs:16]: This is required to make it a hybrid package and we can export both versions cjs and ESModules

**Comment by joheredi** [comment]
> what does this file packages/typespec-ts/src/modular/buildStaticFiles.ts do ?

good catch, it doesn't do anything. I have removed it

File: packages/typespec-ts/src/modular/static/scripts/renameOutput.mjs (hunk 3 above)
Comment (joheredi) [reply packages/typespec-ts/src/modular/static/scripts/renameOutput.mjs:16]: Otherwise, users would be forced to migrate their projects to ESModules

Comment (qiaozha) [reply packages/typespec-ts/src/modular/static/scripts/renameOutput.mjs:16]: get it

**Review by qiaozha — APPROVED** [review]
LGTM

**Pull request merged**